install/kubernetes/helm/istio/templates/operator-configmap.yaml

```diff
@@ -0,0 +1,1241 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: istio-operator
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app: {{ template "istio.name" . }}
+    chart: {{ template "istio.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+data:
+  config: |-
+    apiVersion: apiextensions.k8s.io/v1beta1
+    kind: CustomResourceDefinition
+    metadata:
+      name: virtualservices.networking.istio.io
+      labels:
+        app: istio-pilot
+        chart: istio
```

Comment:
For new install charts: please stop adding 'chart/heritage/release' labels.

I checked helm doc - and it doesn't appear to be any requirement to add them. Labels create secondary indexes in etcd and are expensive, plus label changes usually result in upgrade failures. 

Too late to remove them if they got added in 1.0 - put for new templates we can stop this. ( we should go over the diffs from 1.0 and remove new labels as well )

Reply:
right - not sure why the labels were added in the first place...  Attempting to keep the code base  consistent.  As you point out, that may not be the optimal approach here.  The tiller process adds those labels - perhaps the original PRs that added those labels were meant to ensure the `helm template` rendering was the same as `helm install`.   I'll take a look at removing these labels once we have a 1.1 install that works properly with the additional CRDs.  I seem to recall they were added in the 1.1 series..